packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -161,28 +157,33 @@ export class ToastQueue<T> {
       this.queue.splice(index, 1);
     }
 
-    this.updateVisibleToasts(index);
+    this.updateVisibleToasts({action: 'close', key});
   }
 
   /** Removes a toast from the visible toasts after an exiting animation. */
   remove(key: string) {
-    let index = this.queue.findIndex(t => t.key === key);
-    this.visibleToasts = this.visibleToasts.filter(t => t.key !== key);
-    this.updateVisibleToasts(index);
+    this.queue = this.queue.filter(t => t.key !== key);
+    this.updateVisibleToasts({action: 'remove', key});
   }
 
-  private updateVisibleToasts(oldIndex = -1) {
+  private updateVisibleToasts(options: {action: 'add' | 'close' | 'remove', key?: string}) {
+    let {action, key} = options;
     let toasts = this.queue.slice(0, this.maxVisibleToasts);
-    if (this.hasExitAnimation) {
-      let prevToasts: QueuedToast<T>[] = this.visibleToasts
-          .filter(t => !toasts.some(t2 => t.key === t2.key))
-          .map(t => ({...t, animation: 'exiting'}));
 
-      if (oldIndex !== -1) {
-        toasts.splice(oldIndex, 0, prevToasts?.[0]);
-      }
-
-      this.visibleToasts = toasts;
+    if (action === 'add' && this.hasExitAnimation) {
+      let prevToasts: QueuedToast<T>[] = this.visibleToasts
+        .filter(t => !toasts.some(t2 => t.key === t2.key))
+        .map(t => ({...t, animation: 'exiting'}));
+      this.visibleToasts = prevToasts.concat(toasts).sort((a, b) => b.priority - a.priority);
```

Comment:
This should bring it back exiting animations theoretically, testing and seeing if I can simplify further